rewrite-maven/src/main/java/org/openrewrite/maven/UpgradePluginVersion.java

```diff
@@ -0,0 +1,190 @@
+ */
+@Incubating(since = "7.7.0")
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class UpgradePluginVersion extends Recipe {
+    @Option(displayName = "Group",
+            description = "The first part of a dependency coordinate 'org.openrewrite.maven:rewrite-maven-plugin:VERSION'.",
+            example = "org.openrewrite.maven")
+    String groupId;
+
+    @Option(displayName = "Artifact",
+            description = "The second part of a dependency coordinate 'org.openrewrite.maven:rewrite-maven-plugin:VERSION'.",
+            example = "rewrite-maven-plugin")
+    String artifactId;
+
+    @Option(displayName = "New version",
+            description = "An exact version number, or node-style semver selector used to select the version number.",
+            example = "29.X")
+    String newVersion;
+
+    @Option(displayName = "Version pattern",
+            description = "Allows version selection to be extended beyond the original Node Semver semantics. So for example," +
+                    "Setting 'version' to \"25-29\" can be paired with a metadata pattern of \"-jre\" to select Guava 29.0-jre",
+            example = "-jre",
+            required = false)
+    @Nullable
+    String versionPattern;
+
+    @Option(displayName = "Trust parent POM",
+            description = "Even if the parent suggests a version that is older than what we are trying to upgrade to, trust it anyway. " +
+                    "Useful when you want to wait for the parent to catch up before upgrading. The parent is not trusted by default.",
+            required = false)
+    @Nullable
+    Boolean trustParent;
+
+    @SuppressWarnings("ConstantConditions")
+    @Override
+    public Validated validate() {
+        Validated validated = super.validate();
+        if (newVersion != null) {
+            validated = validated.and(Semver.validate(newVersion, versionPattern));
+        }
+        return validated;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Upgrade Maven plugin version";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Upgrade the version of a plugin using Node Semver advanced range selectors, " +
+                "allowing more precise control over version updates to patch or minor releases.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new UpgradePluginVersionVisitor();
+    }
+
+    private class UpgradePluginVersionVisitor extends MavenVisitor {
+        private final VersionComparator versionComparator;
+        @Nullable
+        private Collection<String> availableVersions;
+
+        public UpgradePluginVersionVisitor() {
+            //noinspection ConstantConditions
+            versionComparator = Semver.validate(newVersion, versionPattern).getValue();
+        }
+
+        @Override
+        public Maven visitMaven(Maven maven, ExecutionContext ctx) {
```

Comment:
I implemented an applicability test which simply checks for the existance of the desired plugin to upgrade. I fully expect for this entire visitor to be cleaned up to be more efficient, though the API for it's usage (`options`) should be stable, and it does work (assuming you aren't trying to get a version from pluginManagement, etc.). But, it does work. Forgive me for this, but I'm going to merge it knowing full-well it could use more love 😬